packages/db/src/query/sensor.ts

```diff
@@ -351,11 +371,95 @@ export async function insertSensorData(data: { sensorId: string; value: number;
         await trx.insert(sensorData).values({
             sensorId: dbSensor.id,
             value: newValue,
-            timestamp: sql<Date>`NOW()`,
+            timestamp: data.timestamp,
         });
     });
 }
 
+/**
+ * Insert out data if the sensor supports it
+ * its the same code as in function and redundant but we keep it so we can change in the future if needed
+ */
+async function insertSensorOutValue(data: SensorDataInput) {
+    await db.transaction(async (trx) => {
+        const dbSensors = await trx.select().from(sensor).where(eq(sensor.id, data.sensorId));
+
+        if (dbSensors.length === 0) {
+            throw new Error("Sensor not found");
+        }
+        const dbSensor = dbSensors[0];
+
+        const lastEntries = await trx
+            .select()
+            .from(sensorDataOutTable)
+            .where(and(eq(sensorDataOutTable.sensorId, dbSensor.id), lt(sensorDataOutTable.timestamp, data.timestamp)))
+            .orderBy(desc(sensorDataOutTable.timestamp))
+            .limit(1);
+
+        if (lastEntries.length === 0) {
+            const newValue = data.value;
+            if (newValue <= 0) {
+                return;
+            }
+            await trx.insert(sensorDataOutTable).values({
+                sensorId: dbSensor.id,
+                value: newValue,
+                timestamp: data.timestamp,
+            });
+            return;
+        }
+        const lastEntry = lastEntries[0];
+
+        const newValue = data.sum ? data.value + lastEntry.value : data.value;
+        if (newValue <= 0 || newValue < lastEntry.value) {
+            return;
+        }
+
+        // in this check we allow 0.4 kwh per minute
+        // so for 15 seconds which is currently the sensor rate we allow 0.1 kwh
+        // in an hour this would be 24 kwh
+        // this is a very high value and should never be reached
+        // but is hopefully a good protection against faulty sensors
+        const timeDiff = new Date(new Date().toUTCString()).getTime() - lastEntry.timestamp.getTime() / 1000 / 60;
+        if (newValue - lastEntry.value > timeDiff * 0.4) {
+            throw new Error("value/too-high");
+        }
+
+        await trx.insert(sensorDataOutTable).values({
+            sensorId: dbSensor.id,
+            value: newValue,
+            timestamp: data.timestamp ? new Date(Number(data.timestamp)) : sql<Date>`NOW()`,
+        });
+    });
+}
+
+/**
+ * Insert a new current value, here a less checks required
+ */
+async function insertSensorCurrentValue(data: SensorDataInput) {
+    await db.insert(sensorDataCurrentTable).values({
+        sensorId: data.sensorId,
+        value: data.value,
+        timestamp: data.timestamp,
+    });
+}
+
+/**
+ * Insert sensor data
+ */
+export async function insertSensorData(data: SensorDataInput) {
+    const promises: Promise<void>[] = [];
+    promises.push(insertSensorInValue(data));
+    if (data.valueOut) {
+        promises.push(insertSensorOutValue(data));
```

Comment:
Können wir das nicht besser alles in eine Tabelle schreiben? Also quasi `sensor_data` um sowas wie `out_value` und `current_value` erweitern?
Dann sparen wir uns zwei separate Tabellen. Dadurch spart man sich auch einiges an Speicherplatz, weil man mit drei Tabellen auch direkt den dreifachen Speicherplatz pro Sensor-Wert verbraucht.